File: java/src/com/nrkei/microservices/rapids_rivers/PacketProblems.java
package com.nrkei.microservices.rapids_rivers;
/*
 * Copyright (c) 2016 by Fred George
 * May be used freely except for training; license required for training.
 * @author Fred George
 */

// Understands issue that arose when analyzing a JSON message
public class PacketProblems {

    private final String originalJson;

    public PacketProblems(String originalJson) {
        this.originalJson = originalJson;
    }

    public boolean hasErrors() {
        return false;
    }
}

File: csharp_v2_beta/src/MicroServiceWorkshop/PacketProblems.cs
/* 
 * Copyright (c) 2016 by Fred George
 * May be used freely except for training; license required for training.
 */

namespace MicroServiceWorkshop
{
    // Understands various information about a received JSON packet
    public class PacketProblems
    {
        private readonly string _originalJson;

        public PacketProblems(string originalJson)
        {
            _originalJson = originalJson;
        }

        public bool HasProblems()
        {
            return false;
        }
    }
}